### ISSUE TITLE
Bug in subgoal_visualization_debug

### ISSUE THREAD

**Issue by rm-2278** [rm-2278/Hieros#26]
I am trying to use the subgoal_visualization_debug function, but I get the error:
Exception in hieros policy: The expanded size of the tensor (128) must match the existing size (8) at non-singleton dimension 2.  Target sizes: [8, 1, 128].  Tensor sizes: [64, 8, 8]

The only part of code to be edited is here:
# Cache actions for debug visualization
                if self._config.subgoal_debug_visualization:
                    cached_actions = torch.cat(
                        (
                            torch.zeros(
                                (
                                    self._config.max_hierarchy - len(cached_actions) - 1,
                                    self._config.envs["amount"],
                                    *self._config.subgoal_shape,
                                ),
                                device=self._config.device,
                            ),
                            cached_actions.detach(),
                        )
                    )
                    self._action_cache[self._subgoal_cache_idx] = [
                        action.detach().unsqueeze(0) for action in cached_actions
                    ]
                    
                    # Compute and cache subgoal rewards for each subactor
                    subgoal_rewards = []
                    for i, (subactor, subactor_state, cached_subgoal) in enumerate(
                        zip(reversed(self._subactors[:-1]), reversed(state[:-1]), cached_subgoals)
                    ):
                        if subactor._compute_subgoal:
                            batch_size = next(iter(subactor_state[0].values())).shape[0]
                            

                            # 1. [B, F] -> [B, 1, F]
                            state_with_time = {k: v.unsqueeze(1) for k, v in subactor_state[0].items()}
                            subgoal_with_batch = cached_subgoal.unsqueeze(0).expand(batch_size, *cached_subgoal.shape) 

                            # 2. Compute subgoal reward
                            reward = subactor._subgoal_reward(state_with_time, subgoal_with_batch)
                            
                            # 3. [B, 1] -> [B]
                            reward_tensor = reward.detach().cpu()
                            subgoal_rewards.append(reward_tensor)
                        else:
                            # If subgoal not computed, store None or zeros
                            subgoal_rewards.append(None)
                    self._subgoal_reward_cache[self._subgoal_cache_idx] = subgoal_rewards

I want you to create a function, test or add lines to debug this bit of code in hieros/hieros.py
